src/components/Header.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {
+  FC,
+  Dispatch,
+  useEffect,
+  useRef,
+  useState,
+  SetStateAction,
+} from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../types/Todo';
+import { Errors } from '../types/Errors';
+
+import { USER_ID } from '../api/todos';
+
+type Props = {
+  todos: Todo[];
+  isAddingTodo: boolean;
+  errorMessage: Errors;
+  setErrorMessage: Dispatch<SetStateAction<Errors>>;
+  handleAddTodo: (
+    { title, userId, completed }: Omit<Todo, 'id'>,
+    setTitle: Dispatch<SetStateAction<string>>,
+  ) => void;
+  handleUpdateTodo: (todoToUpdate: Todo) => void;
+  isLoading: boolean;
+};
+export const Header: FC<Props> = ({
+  todos,
+  setErrorMessage,
+  handleAddTodo,
+  isAddingTodo,
+  errorMessage,
+  handleUpdateTodo,
+  isLoading,
+}) => {
+  const [title, setTitle] = useState('');
+
+  const areTodosAllCompleted = todos.every(todo => todo.completed);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos, errorMessage]);
```

Comment:
Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.